alonzo/formal-spec/protocol-parameters.tex

```diff
@@ -1,98 +1,93 @@
 
 \subsection{Determinism of Script Evaluation}
 \label{sec:determinism}
 
 The data that is passed to the interpreter
-includes the validator, redeemer, information about the transaction carrying
-the script, and some ledger data or protocol parameters.
-It is necessary to maintain a predictable (deterministic) validation outcome over the period between transaction
-submission and script processing.
+includes the validator script, the redeemer, information about the transaction that
+embeds the script, any relevant ledger data, and any relevant protocol parameters.
+It is necessary for the validation outcome of any scripts to remain the same during the entire
+period between transaction
+submission and completion of the script processing.
 %
 In order to achieve this,
-any data that is passed to the interpreter must be
-identical to the data that was provided when the transaction that carries the script was
-constructed.
-Because of this requirement, the carrying transation thus includes a hash of any such data.
-When the transaction is processed, as part of the UTXOW rule, this hash is compared with a hash of the equivalent data. Thus, scripts are only executed if they will be executed with the correct data.
+any data that is passed to the interpreter must be determined by the transaction itself.
+The transaction therefore includes a hash of any data that is not determined by the transaction or the UTXO.
+When the transaction is processed, as part of the UTXOW rule, this hash is compared with a hash of the data that is passed to the interpreter. This
+ensures that scripts are only executed if they have been provided with the intended data.
 
 The $\fun{hashLanguagePP}$ function (Figure~\ref{fig:defs:protocol-parameters}) selects the protocol parameters that are relevant to
 a given set of languages and computes their hash.
 %
-At this time, the only parameter that is passed to the interpreter is the cost model.
+At the time of writing, the only parameter that needs to be passed to the interpreter is the cost model.
 
 \subsection{Script Evaluation Cost Model and Prices}
 \label{sec:cost-mod}
 
-A cost model is used to convert resource primitives into the
-more abstract $\ExUnits$. This conversion is done by the interpreter executing the script,
-which means we can keep the cost model abstract in this specification.
-The actual cost models are recorded in the protocol parameter $\var{costmdls}$.
-%
-By having distinct cost models for each version and changing the conversion coefficients, we can discourage users from
-paying into scripts that have been made using old versions of Plutus, by making these more expensive to execute.
+To convert resource primitives into the
+more abstract $\ExUnits$ during script execution a cost model needs to be supplied to the interpreter.
+The cost models required for this purpose are recorded in the $\var{costmdls}$ protocol parameter.
```

Comment:
i think it makes sense to reiterate that the cost models can be varied independently for each language. I am not sure removing the above statement  
"By having distinct cost models for each version and changing the conversion coefficients, we can discourage users from
paying into scripts that have been made using old versions of Plutus, by making these more expensive to execute." 

is needed

Reply:
There is some explanation about cost models in the introduction, and I've made the point that it is language-specific more explicit there. But I don't like that sentence, because it sounds like we're gonna deprecate everyone's scripts all the time and make it expensive. There's a sentence that we can decrease the costs via the cost model, if we have a more efficient interpreter for example, which of course implies that we could also raise them for whatever reason. But I'm not sure that we should spell that out.